### DISCUSSION TITLE
Gosund SP1 v23 Module no automatic timer when manually switched off

### DISCUSSION THREAD

**Discussion by maximinus333** [arendst/Tasmota#10748] [Q&A] [answered]
I intend using a Gosund Smart Plug 16A flashed as Gosund SP1 v23 Module with Tasmota 8.1.0.2 for switching a heater element 2k W in a hot water boiler. I programmed the timer on every day 6:00 and off 8:30. When I need more hot water it is required to 
switch it on manually during the day when I switch it off later that day it will not switch on again in the set timer intervall as it remains off.
How can I programm it to switch it on manually to remain switched on for 2 hours, switch off automatically and again on and off per time set in the timer configuration
I fiddeled around with PowerOnState 5 and PulseTime1 but no success until now.
Can anybody shed light on this problem?
Regards Gerhard

**Comment by sfromis** [comment, marked as answer]
In the timer configuration screen, you can have a long list (16) of times for on and off, without using rules. PulseTime is mostly useful when you want the same on-period every time, but then it can save you from having the off times in the timer configuration. PulseTime5 is for a different purpose, when you want a fixed off-period and turning on again after the period. For more complex setups, you can also use rules and RuleTimer events.

**Reply by maximinus333** [reply]
Thank you for fast response.
PulseTime5 is for a different purpose, when you want a fixed off-period and turning on again after the period
 Could you give me an example how to obtain exactly the reverse function: 
 I want a fixed on-period after pushing a button turning off again after the period?
Regards Gerhard

**Reply by maximinus333** [reply]
I got it.
Poweronstate 3 and Pulstime xxx
does exactly what I want.
Thank you

**Reply by barbudor** [reply]
That's the normal usage of PulsteTime when not using PowerOnState 5
With all other PowerOnState values, PulseTime will make a Power ON last for a given time and then return to OFF automatically
With PowerOnState 5, PulseTime works inversed and from an ON state, a Power OFF will last for the given duration and automatically switch off.

**Reply by maximinus333** [reply]
@barbudor 
One problem I encounterd. With PulseTime 460 the programmed time every day on 6:00 and off 8:30 is also switched off after 460 instead of  8:30. How can I avoid this?

**Reply by sfromis** [reply]
PulseTime specifically instructs Tasmota to switch off after the given interval. If you want to control the off time-of-day yourself, do not use PulseTime.

**Reply by maximinus333** [reply]
OK, thank you.
